File: data-aggregator/services/inventory.rb
require_relative 'base'

class Inventory < Base
  def initialize(base_url = "http://inventory")
    super(base_url)
  end

  def get_quantity(id)
    get_json("").select{|item| item["id"] == id}.first["quantity"]
  end

  def get_product_name(id)
    get_json("").select{|item| item["id"] == id}.first["name"]
  end
end

File: python-data-aggregator/services/inventory.py
from services.base import Base


class Inventory(Base):
    def get_data(self):
        return self.get("/")

    def get_quantity(self, prod_id):
        return [el for el in self.get_data() if el["id"] == prod_id][0]["quantity"]

    def get_product_name(self, prod_id):
        return [el for el in self.get_data() if el["id"] == prod_id][0]["name"]

